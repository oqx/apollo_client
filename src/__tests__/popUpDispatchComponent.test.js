import React from "react";
import PopUpDispatchComponent from "../components/popUpDispatchComponent";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { completeShow } from "../mockResults";

let store, container, mountedContainer;

const spy = jest.fn();

const handleOnMount = obj => spy();

describe("<PopUpDispatchComponent />", () => {
  beforeEach(() => {
    container = shallow(
      <PopUpDispatchComponent event={completeShow} dispatch={handleOnMount} />
    );
    mountedContainer = mount(
      <PopUpDispatchComponent event={completeShow} dispatch={handleOnMount} />
    );
  });

  it("should render successfully", () => {
    expect(container.length).toEqual(1);
  });

  it("should match snapshot", () => {
    expect(toJson(container)).toMatchSnapshot();
  });

  it("should trigger a function when componentDidMount", () => {
    expect(spy).toHaveBeenCalledTimes(2);
  });

  afterEach(() => {
    spy.mockClear();
  });
});
