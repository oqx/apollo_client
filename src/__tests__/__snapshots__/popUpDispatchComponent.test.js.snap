// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopUpDispatchComponent /> should match snapshot 1`] = `
<div
  key="243426"
/>
`;
